### ISSUE TITLE
RRUL and other tests for bufferbloat 

### ISSUE THREAD

**Issue by brunogm0** [librespeed/speedtest#239]
## Description
The current way of testing is not enough to really diagnose the condition under bufferbloat.
Then if ping and jitter are always measured/displayed the actual latency under load can be shown.
## Why it should be implemented
This library will give an accessible and realistic network test.  The cerowrt blog or bufferbloat.net has many cases showing why the naive test of ping and jitter only at idle is not accurate.  Here there are many implementations. https://www.bufferbloat.net/projects/codel/wiki/RRUL_test_suite/
https://www.bufferbloat.net/projects/bloat/wiki/Tests_for_Bufferbloat/
https://www.bufferbloat.net/projects/bloat/wiki/RRUL_Spec/
## Optional: implementation suggestions
http://www.dslreports.com/speedtest
https://medium.com/netflix-techblog/building-fast-com-4857fe0f8adb

**Comment by adolfintel** [comment]
You're not the first to request a buffer bloat test, I tried to implement it last year but I did not release it.
Personally I consider this to be a pretty useless measurement, because it involves too many factors: number of streams up and down, how the browser schedules the connections, the network driver, how the network card schedules the packets, buffer sizes, how your modem allocates the available bandwidth, and so on and so forth.
Basically, you get a different result on different machines, sometimes even on consecutive tests. This also happens (for me at least) on that dslreports test that you linked, it's basically a random number generator between my ping and infinity.

**Comment by robingroppe** [comment]
What you described as a useless measurement is exactly what the bufferbloat tests purpose is.
It gives you an idea how well your scheduler at your home router works if at all, if there is an issue behind your router or in front of it. 
With reasonable QoS you can get pretty stable results. That said, most of home network equipment doesn't care and enterprise equipment wants to be configured properly.

I'd like to ask you to overthink your decision to drop that idea.
